src/viztracer/modules/vcompressor/vc_dump.c

```diff
@@ -161,14 +162,169 @@ int dump_parsed_trace_events(PyObject* trace_events, FILE* fptr)
             goto clean_exit;
         }
     }
+
+    // Iterate through instant events
+    ppos = 0;
+    while (PyDict_Next(instant_events, &ppos, &key, &value)) {
+        uint64_t pid = PyLong_AsLong(PyTuple_GetItem(key, 0));
+        uint64_t tid = PyLong_AsLong(PyTuple_GetItem(key, 1));
+        const char* name = PyUnicode_AsUTF8(PyTuple_GetItem(key, 2));
+        char scope = PyUnicode_AsUTF8(PyTuple_GetItem(key, 3))[0];
+        fputc(VC_HEADER_INSTANT_EVENTS, fptr);
+        fwrite(&pid, sizeof(uint64_t), 1, fptr);
+        fwrite(&tid, sizeof(uint64_t), 1, fptr);
+        fwritestr(name, fptr);
+        fputc(scope, fptr);
+        if (write_instant_args(value, fptr) != 0) {
+            goto clean_exit;
+        }
+    }
+clean_exit:
+    if (PyErr_Occurred()) {
+        return 1;
+    }
+
+    return 0;
+}
+
+int write_instant_args(PyObject* instant_args, FILE* fptr) {
```

Comment:
The major way we "compress" data in `vcompressor` is to reduce the redundancy, especially when multiple data entries have the same info except for ts. However, for instant events, it seems like we can't do that easily because each instant event could potentially have completely different data. In that case, the method to rearrange data like this would only save us the redundant fields like `ph` or `cat`, not the real data.

So, to make this simpler, considering the manual rearrangement of data was not that helpful, maybe just dump the whole thing as binary? For each instant data with the same pid&tid&name&scope, dump the dict you kept as json(plus zlib?), the code should be simpler(no loops in `write_instant_args` function anymore) and the compression rate should be better(with zlib of course).

The code to do it is basically in `dump_file_info`. Most of the code in that function is to make `zlib` + `json` work. I'm thinking maybe we could separate the `zlib` + `json` part out as an utility function - which could be called from both dumping file info and dumping instant events.

The good thing about this is that we can keep one less compressing method - the current way you compress instant events is unique and probably won't be applicable to other cases(or for the cases it's applicable, they could/should be replaced with `json` + `zlib`).